File: 2nd Semester/02. Data Structure Lab/Lab Class Code/Lab 03 -- 01-02-25/Class Work/linear search.c
#include <stdio.h>
int main()
{

    int i, loc, n,c;
    float a[50], item;
    printf("No of Element:");
    scanf("%d", &n);
    for(int i=0; i<n; i++)
    {

        printf("\na[%d]=", i);
        scanf("%f", &a[i]);

    }
    for(;;)
    {
        loc = -1;
        printf("Item:");
        scanf("%f", &item);

        for(int i=0;i<n;i++)
        {

            if(item==a[i])
            {
                loc = i;
                break;
            }
        }
        if(loc == -1)
        {
            printf("\nabsent");
        }
        else
        {
            printf("\n%f found in %d position.", item, loc);
        }
        printf("Press 0 to exit, anything else to continue");
        scanf("%d", &c);
        return 0;
    }
    return 0;
}

File: 3rd Semester/02. OOP Lab/Lab Report/Lab Report 01/src/linearSearch.java
public class linearSearch {
    public static void main(String[] args){
        int[] a = {10,20,30,40,50};
        int item = 40;
        int count = 0;
        boolean t = true;

        for(int x : a){
            if(x == item){
                t = false;
                break;
            }
        }
        //System.out.println((t) ? "Item Not Found :(" : "Item Found :)");
        if(t){
            System.out.println("Item Not Found :(");
        }
        else{
            System.out.println("Item Found :)");
        }
    }
}
